src/Storages/MergeTree/MergeTreeSink.cpp

```diff
@@ -155,18 +155,12 @@ void MergeTreeSink::finishDelayedChunk()
                 {
                     ProfileEvents::increment(ProfileEvents::DuplicatedInsertedBlocks);
                     LOG_INFO(storage.log, "Block with ID {} already exists as part {}; ignoring it", block_id, res.first.getPartName());
-                }
-                else
-                {
-                    added = storage.renameTempPartAndAdd(part, transaction, partition.temp_part.builder, lock);
-                    transaction.commit(&lock);
+                    continue;
```

Comment:
This is incorrect, because of `continue` - the code below (`if (added)`) will not be executed now, but previously it was.

Reply:
But if you go to this logic "if (!res.second)" "added" must be false, so the following logic does not need to be executed